argocd_sale/models/subscription.py

```diff
@@ -111,3 +135,38 @@ def cron_update_payment_provider_subscriptions(self):
             late_subs.close_subscription()
             late_subs._do_grace_period_action()
         return res
+
+    def _get_free_period(self):
+        self.ensure_one()
+        existing_subs = self.partner_id.subscription_ids
+        if self.partner_id.parent_id:
+            existing_subs += self.partner_id.subscription_ids
```

Comment:
Shouldn't this be existing_subs += self.partner_id.parent_id.subscription_ids. Currently parent_id is missing